tests/helpers/optimized_video_fixtures.py

```
@@ -71,9 +73,22 @@ def get_cached_or_create(cache_key: str, factory_func, *args, **kwargs):
     """
-    if cache_key not in _session_cache:
-        _session_cache[cache_key] = factory_func(*args, **kwargs)
-    return _session_cache[cache_key]
+    cached = _session_cache.get(cache_key)
+
+    if isinstance(cached, models.Model):
+        pk = getattr(cached, "pk", None)
+        if pk is not None and cached.__class__.objects.filter(pk=pk).exists():
+            # Return a fresh instance to avoid stale relations after DB flushes.
+            return cached.__class__.objects.get(pk=pk)
+        # Stale instance – drop it so we fall back to factory creation.
```

Ruff (0.14.0)

Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)